src/main/java/io/vavr/jackson/datatype/serialize/VavrSerializers.java

```diff
@@ -110,7 +110,7 @@ public VavrSerializers(VavrModule.Settings settings) {
             return new LazySerializer(type);
         }
         if (Option.class.isAssignableFrom(raw)) {
-            return new OptionSerializer(type, settings.useOptionInPlainFormat());
+            return new OptionSerializer(type.getContentType(), settings.useOptionInPlainFormat());
```

Comment:
In my opinion, yes. But I didn't have time to check other types of Vavr. I will do it today or tomorrow and - test scenarios and fixes.